fluent-react/README.md

```diff
@@ -0,0 +1,129 @@
+# fluent-react
+
+`fluent-react` provides React bindings for Project Fluent, a localization
+framework designed to unleash the expressive power of the natural language.
+
+
+## Installation
+
+`fluent-react` can be used both on the client-side and the server-side.  You
+can install it from the npm registry or use it as a standalone script (as the
+`FluentReact` global).
+
+    npm install fluent-react
+
+
+## How to use
+
+`fluent-react` takes advantage of React's Components system and the virtual
+DOM.  Translations are stored using the Fluent syntax ([FTL][]) and exposed to
+components via the _provider_ pattern.
+
+An example FTL file might look like the following:
+
+```
+hello-world = Hello, world!
+welcome = Welcome, $username.
+```
+
+
+Wrap your localizable elements in `LocalizedElement`:
+
+```javascript
+import { LocalizedElement } from 'fluent-react';
+
+<LocalizedElement id="hello-world">
+    <p>Hello, world!</p>
+</LocalizedElement>
+```
+
+The `id` prop should be the unique identifier of the translation.  Any
+attributes found in the translation will be applied to the wrapped element.
+You can also pass arguments to the translation as $-prefixed props on
+`LocalizedElement`:
+
+```javascript
+<LocalizedElement id="welcome" $username={name}>
+    <p>{'Welcome, {$username}'}</p>
+</LocalizedElement>
+```
+
+Notice how in the above example the text content of the `<p>` element is
+a string expression, not a text node.  It is provided as help for the developer
+and may be used in the future to extract the source copy. For now, however,
+it's not used for anything.  By putting the text in a simple string expression
+we're saving React's cycles: `<p>Welcome, { $username }</p>` would be rendered
+by React before being passed to `LocalizedElement` where it would be replaced
+by the translation anyways.
+
+It's also possible to pass React elements as arguments to translations:
+
+```javascript
+<LocalizedElement
+    id="welcome"
+    $linkA={
+        <a href="http://example.com/A">A</a>
+    }
+    $linkB={
+        <LocalizedElement id="link-to-b">
+            <a href="http://example.com/B">B</a>
+        </LocalizedElement>
+    }>
+    <p>{'Click { $linkA } or { $linkB}.'}</p>
+</LocalizedElement>
+```
+
+All `<LocalizedElement> components need access to the central localization
+store so they can subscribe to it.  The `<LocalizationProvider>` component
+implements the _provider_ pattern known from Redux.  Taking advantage of
+React's [context][] feature, it makes the translation store available to all
+localized elements in the app without passing it explicitly.
+
+```javascript
+import 'fluent-intl-polyfill';
+import negotiateLanguages from 'fluent-langneg';
+import { LocalizationProvide } from 'fluent-react';
+
+ReactDOM.render(
+  <LocalizationProvider
+    locale={negotiateLanguages(navigator.language, ['en-US', 'de'])}
```

Comment:
yeah, it should be. I documented it in NOTES as something I want to do. I'll probably wait for `fluent-langneg`.